src/session/saveSessionDialogue.component.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from '@mui/material';
+import React, { useState } from 'react';
+import { useAppSelector } from '../state/hooks';
+import { useSaveSession } from '../api/sessions';
+
+export interface SessionDialogueProps {
+  open: boolean;
+  onClose: () => void;
+  sessionName: string | undefined;
+  sessionSummary: string;
+  onChangeSessionName: (sessionName: string | undefined) => void;
+  onChangeSessionSummary: (sessionSummary: string) => void;
+}
+
+const SaveSessionDialogue = (props: SessionDialogueProps) => {
+  const {
+    open,
+    onClose,
+    sessionName,
+    sessionSummary,
+    onChangeSessionName,
+    onChangeSessionSummary,
+  } = props;
+
+  const state = useAppSelector(({ config, ...state }) => state);
+  const { mutateAsync: saveSession } = useSaveSession();
+
+  const [nameError, setError] = useState(false);
+  const [errorMessage, setErrorMessage] = React.useState<string | undefined>(
+    undefined
+  );
+  const handleClose = React.useCallback(() => {
+    onClose();
+    onChangeSessionName(undefined);
+    onChangeSessionSummary('');
+  }, [onClose, onChangeSessionName, onChangeSessionSummary]);
+
+  const handleExportSession = React.useCallback(() => {
+    if (sessionName) {
+      const session = {
+        name: sessionName,
+        session_data: state,
+        summary: sessionSummary,
+        auto_saved: false,
+      };
+      saveSession(session)
+        .then((response) => handleClose())
+        .catch((error) => {
+          setError(true);
+          console.log(error.message);
+          setErrorMessage(error.message);
+        });
+    } else {
+      setError(true);
+      setErrorMessage('Please enter a name');
+    }
+  }, [handleClose, saveSession, sessionName, sessionSummary, state]);
+
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth="lg">
+      <DialogTitle>Save Session</DialogTitle>
+      <DialogContent>
+        <TextField
+          label="Name*"
```

Comment:
Minor point - but the `*` shouldn't be added to the `label` - instead add the `required` prop to the `TextField` - this adds the `*` automatically without then messing up accessibility (i.e. the * is marked with aria-hidden so screen readers ignore it, etc.)